main.js

```diff
@@ -0,0 +1,55 @@
+/*variables for input fields and the grow tree button*/
+
+var button = document.getElementById("grow")
+var heightVal = document.getElementById("height")
+var characterVal = document.getElementById("character")
+
+/*event lsitener to detect when the grow tree button is pushed*/
+button.addEventListener("click", userInputForTree)
+
+/*this function is used to get the initial user input.*/
+function userInputForTree() {
+  var rows = heightVal.value;
+  var width = characterVal.value;
+  checkForProperFormat(rows, width);
+}
+
+/*checking that each input field has a value*/
+function checkForProperFormat(rows, width) {
+  if(!rows || !width) {
+    alert("Both input fields need a value!");
+  } else {
+    buildTree(rows, width);
+  }
+}
+
+/*function for building the tree from the user input. Then calls the treeDisplay function*/
+function buildTree(rows, width) {
```

Comment:
I like the explicit nature and the breaking down things into components that you've done for the most part, but I think you may have taken it a little too far on this one.  :)  Having a function that only calls another function is probably just unnecessary code that will/could break, and another thing that has to be changed every time you change the parameters of the other function.  I'd suggest calling `treeDisplay` directly from line 22.